src/main/ruby/core/hash.rb

```diff
@@ -491,4 +491,27 @@ def transform_values!
     end
     self
   end
+
+  def transform_keys
+    return to_enum(:transform_keys) { size } unless block_given?
+
+    h = {}
+    each_pair do |key, value|
+      h[yield(key)] = value
+    end
+    h
+  end
+
+  def transform_keys!
+    return to_enum(:transform_keys!) { size } unless block_given?
+
+    Truffle.check_frozen
+
+    each_pair do |key, value|
+      new_key = yield(key)
+      delete key unless key == new_key
```

Comment:
`==` seems wrong here (should be `eql?`).
It looks like MRI always delete, so it's probably best to do the same here.

Reply:
Oops, yeah I see why #== is wrong and #eql? is right. Fixed.

I *think* there are edge cases where always deleting isn't right. [This spec](https://github.com/graalvm/truffleruby/blob/ce69384cd7e2e412b83b1a08b475e3dd50794284/spec/ruby/core/hash/transform_keys_spec.rb#L20) seems to show one of those cases, e.g.:

```ruby
{a: 1, b: 2, c: 3}.transform_keys { |_| :a } #=> {:a=>3}
```

Reply:
My specs are hanging for some reason if I don't check equality so I'm not sure which is failing.

Reply:
Running `spec/ruby/core/hash/transform_keys_spec.rb` hangs without the check? That sounds odd.
Maybe due to iterating while deleting elements.
Here is MRI's implementation FWIW: https://github.com/ruby/ruby/blob/2546685dfbdf7397cae5e1db92fd22ec2808cb1d/hash.c#L1947
So it looks like:
```ruby
keys.each do |key|
  new_key = yield(key)
  self[new_key] = delete(key)
end
```

That sounds safer than deleting pairs while iterating over them, could you switch to that?

Reply:
Ah, I see where I went wrong. Thank you! Done.